app/src/main/java/com/github/sdp/tarjetakuna/location/Location.kt

```diff
@@ -0,0 +1,107 @@
+     */
+    fun askForLocationPermission(context: AppCompatActivity) {
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                android.Manifest.permission.ACCESS_COARSE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                context,
+                android.Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            requestPermission(context)
+        }
+    }
+
+    /**
+     * Get the current location of the user if the permission is granted
+     */
+    fun captureCurrentLocation(context: AppCompatActivity) {
+        // Check if the location permission is granted, and if not, request it
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            Log.i("Location", "Permission not granted")
+            // Do nothing since the user rejected the permission
+        } else {
+            if (locationManager == null) {
+                locationManager =
+                    context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+            }
+            locationManager!!.requestLocationUpdates(
+                LocationManager.GPS_PROVIDER,
+                0,
+                0f,
+                locationListener
+            )
+        }
+
+    }
+
+    // What to do when the location changes
+    private val locationListener: LocationListener = LocationListener { location ->
+        val newLocation = Coordinates(location.latitude, location.longitude)
+        if (!currentLocation.isSameCoordAs(newLocation)) {
+            currentLocation = newLocation
+            Log.i(
+                "Location",
+                "Location changed: latitude: ${Location.getCurrentLocation().latitude}, " +
+                        "longitude: ${Location.getCurrentLocation().longitude}"
+            )
+        }
+    }
+
+    /**
+     * Request the location permission
+     */
+    private fun requestPermission(context: AppCompatActivity) {
+        val requestcode = 1
+        ActivityCompat.requestPermissions(
+            context,
+            arrayOf(
+                android.Manifest.permission.ACCESS_FINE_LOCATION,
+                android.Manifest.permission.ACCESS_COARSE_LOCATION
+            ),
+            requestcode
+        )
+    }
+
+    /**
+     * Getter for the current location
+     */
+    fun getCurrentLocation(): Coordinates {
+        return currentLocation
+    }
+
+    /**
+     * Setter for the location manager
+     */
+    fun setLocationManager(manager: LocationManager) {
+        locationManager = manager
+    }

```

Comment:
You can remove the private for the locationManager and remove these two functions

Reply:
What do you mean remove these two functions? Are you talking about getCurrentUser and setLocationManager? If so, these are two different functions that touches different variables

Reply:
True sorry, I saw badly.
Still, you can use this process if you don't want to create new functions : 
```
var someProperty: String = "defaultValue"
    get() = field
    set(value) { field = value }
```
But it's not required, as you wish here ! :)

Reply:
Oh that's interesting I will try it :)